### PR TITLE
feat: cleancode.md

### PR DESCRIPTION
**Pull request by bsrakts**
create

### CODE REVIEW

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+- Okuması zor ile okuması kolay kod arasındaki farkı nasıl ayırt edebilirim?
+- Kodunuzu daha temiz hale getirmek için programlama prensiplerini kullanın.
+- İyi yorumlar yazın.
+
+### Temiz Kod Yazma
+
+Aşağıdaki iki JavaScript kod parçasını inceleyin:
+
+Bu okuması zor ve kötü görünümlü bir kod örneği:
```
Comment (kutaui) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
Bu okuması zor ve kötü görünen bir kod örneği:
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  array.forEach(function(number) {
+    sum += number;
+  });
+  return sum;
+};
+
+sumArray([2, 2, 2]);
+~~~
+
+İnanın veya inanmayın, her iki fonksiyon da aynı şeyi\(tamamen aynı şekilde!\) yapıyor ve her ikisi de tamamen geçerli kodlar, ancak görüldüğü üzere ikincisi takip etmesi daha kolaydır.Başka birisiyle ortak bir projede çalıştığınızı ve ilk fonksiyonu yazdığını hayal edin.. İşinizi yapabilmeniz için ne olup bittiğini anlamalısınız, ilk örnekte bunun için ne kadar süre harcamanız gerekeceğini düşünün. Kendi projenizi yazarken ilk fonksiyonu bir veya iki hafta önce yazdığınızı ve tek başınıza çalıştığınızı düşünün. Tam olarak ne yaptığınızı muhtemelen hatırlamayacaksınız. Her şeyi çözmek ve anlamak için tekrar iyi bir zaman harcamanız gerekecektir.
```
Comment (kutaui) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
İnanın veya inanmayın, her iki fonksiyon da aynı şeyi\(tamamen aynı şekilde!\) yapıyor ve her ikisi de tamamen geçerli kodlar, ancak görüldüğü üzere ikincisini takip etmesi daha kolay.Başka birisiyle ortak bir projede çalıştığınızı ve ilk fonksiyonu yazdığını hayal edin.. İşinizi yapabilmeniz için ne olup bittiğini anlamalısınız, ilk örnekte bunun için ne kadar süre harcamanız gerekeceğini düşünün. Kendi projenizi yazarken ilk fonksiyonu bir veya iki hafta önce yazdığınızı ve tek başınıza çalıştığınızı düşünün. Tam olarak ne yaptığınızı muhtemelen hatırlamayacaksınız. Her şeyi çözmek ve anlamak için tekrar iyi bir zaman harcamanız gerekecektir.
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    // Kötü (bu örnekte fonksiyon bir isimdir)
+    function myName() {
+      return "Thor";
+    }
+    ~~~
+
+
+### Assignment
```
Comment (kutaui) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
### Ödev
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  });
+  return sum;
+};
+
+sumArray([2, 2, 2]);
+~~~
+
+İnanın veya inanmayın, her iki fonksiyon da aynı şeyi\(tamamen aynı şekilde!\) yapıyor ve her ikisi de tamamen geçerli kodlar, ancak görüldüğü üzere ikincisi takip etmesi daha kolaydır.Başka birisiyle ortak bir projede çalıştığınızı ve ilk fonksiyonu yazdığını hayal edin.. İşinizi yapabilmeniz için ne olup bittiğini anlamalısınız, ilk örnekte bunun için ne kadar süre harcamanız gerekeceğini düşünün. Kendi projenizi yazarken ilk fonksiyonu bir veya iki hafta önce yazdığınızı ve tek başınıza çalıştığınızı düşünün. Tam olarak ne yaptığınızı muhtemelen hatırlamayacaksınız. Her şeyi çözmek ve anlamak için tekrar iyi bir zaman harcamanız gerekecektir.
+
+Gerçekten ikincisi takip etmek için çok daha kolaydır. Kodun içerisindeki her şeyi tam olarak anlamasanız bile değişkenler açıkça isimlendirildiği için tahmin edebilirsiniz.Girintili yazılması da fonksiyonun farklı bölümlerini ayırt edebilecek kadar tutarlıdır.
```
Comment (kutaui) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
Gerçekten ikincisini takip etmek çok daha kolaydır. Kodun içerisindeki her şeyi tam olarak anlamasanız bile değişkenler açıkça isimlendirildiği için tahmin edebilirsiniz. Girintinin tutarlı kullanımı da fonksiyonun farklı bölümlerini anlaşılabilir kılmaktadır.
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
+### Giriş
+
+Geliştiriciler kod yazmaktan daha çok kod okumaya zaman harcıyorlar. Bu kendi kodunuzda bile doğrudur.Lütfen kendinize ve kodunuzu kullanacak,bakımını yapacak ve geliştirecek kişilere iyilik yapmak için okunabilir kod yazmayı öğrenin.
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
Geliştiriciler kod yazmaktan daha çok, kod okumaya zaman harcıyorlar. Bu kendi kodunuz için bile geçerlidir. Lütfen kendinize ve kodunuzu kullanacak, bakımını yapacak veya ileride geliştirecek kişilere iyilik yapmak için okunabilir kod yazmayı öğrenin.
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+  array.forEach(function(number) {
+    sum += number;
+  });
+  return sum;
+};
+
+sumArray([2, 2, 2]);
+~~~
+
+İnanın veya inanmayın, her iki fonksiyon da aynı şeyi\(tamamen aynı şekilde!\) yapıyor ve her ikisi de tamamen geçerli kodlar, ancak görüldüğü üzere ikincisini takip etmesi daha kolay.Başka birisiyle ortak bir projede çalıştığınızı ve ilk fonksiyonu yazdığını hayal edin.. İşinizi yapabilmeniz için ne olup bittiğini anlamalısınız, ilk örnekte bunun için ne kadar süre harcamanız gerekeceğini düşünün. Kendi projenizi yazarken ilk fonksiyonu bir veya iki hafta önce yazdığınızı ve tek başınıza çalıştığınızı düşünün. Tam olarak ne yaptığınızı muhtemelen hatırlamayacaksınız. Her şeyi çözmek ve anlamak için tekrar iyi bir zaman harcamanız gerekecektir.
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
İnanın veya inanmayın, her iki fonksiyon da aynı şeyi\(tamamen aynı şekilde!\) yapıyor ve her ikisi de tamamen geçerli kodlar ancak görüldüğü üzere ikincisini takip etmesi daha kolay. Başka birisiyle ortak bir projede çalıştığınızı ve ilk fonksiyonu onun yazdığını hayal edin. İşinizi yapabilmeniz için ne olup bittiğini anlamalısınız, ilk örnekte bunun için ne kadar süre harcamanız gerekeceğini düşünün. Kendi projenizde tek başınıza çalıştığınızı ve bir iki hafta önce ilk örnekteki fonksiyonu yazdığınızı düşünün. Tam olarak ne yaptığınızı muhtemelen hatırlamayacaksınız. Her şeyi çözmek ve anlamak için _yine_ bayağı bir zaman harcamanız gerekecektir.
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+Gerçekten ikincisini takip etmek çok daha kolaydır. Kodun içerisindeki her şeyi tam olarak anlamasanız bile değişkenler açıkça isimlendirildiği için tahmin edebilirsiniz. Girintinin tutarlı kullanımı da fonksiyonun farklı bölümlerini anlaşılabilir kılmaktadır.
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
İkincisini takip etmek gerçekten çok daha kolaydır. Kodun içerisindeki her şeyi tam olarak anlamasanız bile değişkenler açıkça isimlendirildiği için tahmin edebilirsiniz. Girintinin tutarlı kullanımı da fonksiyonun farklı bölümlerini anlaşılabilir kılmaktadır.
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+Harika bir JavaScript kodunu oluşturma yöntemleri hakkında birçok farklı görüş var. En önemli şey sadece tutarlı olmanızdır. Kodları girintilemek için tab kullanan ve boşluk kullanan kullanıcılar arasındaki savaş o kadar derinleşti ki [bu aslında artık bir şaka](https://www.youtube.com/watch?v=SsoOG6ZeyUI), ancak tutarlı olduğunuz sürece gerçekten hangisini tercih ettiğiniz önemli değil.
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
Bir JavaScript kodunu harika kılan şeyin ne olduğu hakkında birçok farklı görüş var. En önemli şey sadece tutarlı olmanızdır. Kodları girintilemek için tab kullanan ve boşluk kullanan kullanıcılar arasındaki savaş o kadar derinleşmiştir ki [artık bir şaka haline gelmiştir](https://www.youtube.com/watch?v=SsoOG6ZeyUI) ancak tutarlı olduğunuz sürece gerçekten hangisini tercih ettiğiniz önemli değil.
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+### Kurallar veya Kılavuzlar
+
+1.  Girintileme: Kullandığınız girintileme stili gerçekten önemli değil. Farklı JS stil rehberleri farklı seçenekleri önerebilir ve biri diğerinden üstün değildir. Her halükarda önemli olan tutarlılıktır. Alıştırmalarımızda girintileme için 2 boşluk kullanacağız.
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
1.  Girintileme: Kullandığınız girintileme stili _gerçekten_ önemli değil. Farklı JS stil rehberleri farklı seçenekler önermektedir ve hiçbiri bir diğerinden üstün değildir. Her halükarda önemli olan tutarlılıktır. Alıştırmalarımızda girintileme için 2 boşluk kullanacağız.
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+
+2.  Noktalı virgüller: Noktalı virgüller, JavaScript'te çoğunlukla isteğe bağlıdır. Çünkü JS derleyicisi atlanmışlarsa otomatik olarak ekleyecektir. Fakat bu fonksiyonalite bazı durumlarda bozulabilir ve kodunuzda hatalara neden olabilir. Bu yüzden noktalı virgülü eklemeye alışmak daha iyidir. Yapmanız gereken sadece eklemek!
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
2.  Noktalı virgüller: Noktalı virgüller, JavaScript'te _çoğu zaman_ isteğe bağlıdır. Çünkü eksikse, JS derleyicisi otomatik olarak ekleyecektir. Fakat bu fonksiyonalite bazı durumlarda bozulabilir ve kodunuzda hatalara neden olabilir. Bu yüzden noktalı virgülü eklemeye alışmak daha iyidir. Ekleyin gitsin!
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+
+3. Satır uzunluğu: Yine farklı stil rehberleri bu konuda farklı önerilerde bulunabilir, ancak neredeyse tümü her satır uzunluğunu sınırlandırmayı önerir. Bu kural diğer kurallar kadar sıkı değildir. Fakat genel bir kural olarak, kodunuzun her satırı yaklaşık 80 karakterden sonra manuel olarak bölmeniz kodunuzu daha okunaklı hale getirir. Birçok kod düzenleyecisi, bu eşii geçtiğinizde size uyarı verecek bir satır göstergesine sahiptir. Satırları manuel olarak bölerken, operatör veya virgülün hemen sonrasında bölmeniz önerilir. Örneğin şunları yapabilirsiniz:
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
3. Satır uzunluğu: Yine farklı stil rehberleri bu konuda farklı önerilerde bulunabilir ancak neredeyse HEPSİ, satır uzunluğunu sınırlandırmayı önerir. Bu kural diğer kurallar kadar sıkı değildir. Fakat genel bir kural olarak, kodunuzun her satırını yaklaşık 80 karakterden sonra manuel olarak bölmeniz kodunuzu daha okunaklı hale getirir. Birçok kod düzenleyicisi, bu eşiği geçtiğinizde size uyarı verecek bir satır göstergesine sahiptir. Satırları manuel olarak bölerken, operatör veya virgülün hemen sonrasında bölmeniz önerilir. Devam satırlarını düzenlemenin birkaç yolu vardır. Örneğin şunları yapabilirsiniz:
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+
+   - devam satırlarını aynı seviye girintiyle girintileyin,
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
   - devam satırlarını bir kez girintileyin.
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+   - devam eden satırları dikey olarak ilk değişkenle hizalayın,
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
   - devam satırlarını, ilk değişkeni baz alarak dikeylemesine hizalayın
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+   - veya tamamen farklı bir biçim kullanabilirsiniz. Kurallar katı değildir ve iş ortamından iş ortamına değişkenlik gösterebilir, yine de tutarlı olmaya dikkat edin.
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
   - veya tamamen farklı bir biçim kullanabilirsiniz. Kurallar katı değildir ve iş ortamından iş ortamına değişkenlik gösterebilir, yeter ki tutarlı olmaya dikkat edin.
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+     oneMoreReallyLongThing;
+
+   // Diğer olası bir format:
+   let anotherReallyReallyLongLine = something + somethingElse + anotherThing +
+                                     howManyTacos + oneMoreReallyLongThing;
+   ~~~
+   
+   ​
+
+4.  Fonksiyonları ve Değişkenleri isimlendirme: Fonksiyonlar ve değişkenler için isimlendirme açık olmalıdır. Her zaman camelCase kullanın. Tutarlılık ve okunabilirlik için değişkenler her zaman bir isim veya sıfatla(yani bir isim öbeğiyle) başlamalı ve fonksiyonlar bir fiil ile başlamalıdır. Döngü veya callback fonksiyonu bağlamında tek karakterli değişken adları kullanmak sorun değil ama başka yerlerde kullanmamaya özen gösterin.
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
4.  Fonksiyonları ve Değişkenleri İsimlendirme: Fonksiyonlar ve değişkenler için isimlendirme açık olmalıdır. Her zaman camelCase kullanın. Tutarlılık ve okunabilirlik için değişkenler her zaman bir isim veya sıfatla (yani bir tamlama oluşturmalı) başlamalı ve fonksiyonlar bir fiil ile başlamalıdır. Döngü veya callback fonksiyonu bağlamında tek karakterli değişken adları kullanmak sorun değil ama başka yerlerde kullanmamaya özen gösterin.
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+      return "Thor";
+    }
+    ~~~
+
+
+### Ödev
+
+<div class="lesson-content__panel" markdown="1">
+
+1.  [Bu temiz kod ipuçlarının listesi](https://onextrapixel.com/10-principles-for-keeping-your-programming-code-clean/).
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
1.  [Temiz koda dair ipucu listesi](https://onextrapixel.com/10-principles-for-keeping-your-programming-code-clean/).
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+2.  [Makale](https://blog.codinghorror.com/coding-without-comments/), [Diğer makale](https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/) kodunuzda yorumların rolü içi bu makalelere göz atabilirsiniz.
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
2.  [Bu makale](https://blog.codinghorror.com/coding-without-comments/) ve [bir de bu makale](https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/) kodunuzdaki yorumların önemini anlatmaktadır.
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+</div>
+
+### Bilgi ölçme
+
+Bu bölüm temiz kod yazma dersinde anladığınızı kendi kendinize kontrol etmek için sorular içermektedir. Bir soruyu cevaplamakta zorlanıyorsanız, üzerine tıklamanız ve bağlantıda verilen materyale göz atmanız önerilir.
+
+- [Temiz kod yazmak neden önemli?](#writing-clean-code)
+- [Önceden bahsedilen 5 temiz kod yazma prensibini adlandırın.](https://onextrapixel.com/10-principles-for-keeping-your-programming-code-clean/)
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
- [Önceden bahsedilen 5 temiz kod yazma prensibini sayın](https://onextrapixel.com/10-principles-for-keeping-your-programming-code-clean/)
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+- [İyi yorumlar ve kötü yorumların farklılıkları nedir?](https://onextrapixel.com/10-principles-for-keeping-your-programming-code-clean/)
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
- [İyi ve kötü yorumların arasındaki farklar nelerdir?](https://onextrapixel.com/10-principles-for-keeping-your-programming-code-clean/)
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+
+### Ek Kaynaklar
+
+Bu bölüm, ilgili içeriğe yönlendiren yararlı bağlantılar içerir. Zorunlu değildir, dolayısıyla ek olarak düşünülebilir.
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
Bu alanda içerikle alakalı faydalı linkler bulunmaktadır. Zorunlu değildir, ek olarak düşünülmelidir.
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+
+* [Güzel bir görüş yazısı](https://www.martinfowler.com/bliki/CodeAsDocumentation.html)
+* Tam rehber: [kendini belgeleyen kodlarla](http://wiki.c2.com/?SelfDocumentingCode)
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
* ["Kendi kendini açıklayan kod" rehberi](http://wiki.c2.com/?SelfDocumentingCode)
```

File: temel_bilgiler/javascript_temelleri/cleancode.md [hunk 22]
```diff
@@ -0,0 +1,126 @@
...
+* [Airbnb stil rehberi](https://github.com/airbnb/javascript)  
+* [Cümleler yazmak için yöntem zincirleme](https://web.archive.org/web/20190211152543/https://javascriptissexy.com/beautiful-javascript-easily-create-chainable-cascading-methods-for-expressiveness/)
```
Comment (ubsoydan) [inline temel_bilgiler/javascript_temelleri/cleancode.md]:
```suggestion
* [Cümle yazmak için zincirleme yöntemini kullanmak](https://web.archive.org/web/20190211152543/https://javascriptissexy.com/beautiful-javascript-easily-create-chainable-cascading-methods-for-expressiveness/)   
```

**Review by ubsoydan — CHANGES_REQUESTED** [review]
Good one!
Considering this is your first one, hopefully many more will come and will shine! 
Thanks for your contribution Büşra!

**Pull request merged**